src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,135 @@
+    }
+
+    resp.data.topics.reserve(request.data.topics->size());
+    auto timeout_ms = request.data.timeout_ms;
+    kafka::error_code resp_error;
+    ss::sstring resp_error_message;
+
+    for (const list_partition_reassignments_topics& topic :
+         *request.data.topics) {
+        auto topic_config = ctx.metadata_cache().get_topic_cfg(
+          model::topic_namespace{model::kafka_namespace, topic.name});
+        if (!topic_config) {
+            vlog(klog.error, "Unknown topic: {}", topic.name);
+            resp.data.error_code = error_code::unknown_topic_or_partition;
+            resp.data.error_message = ss::sstring{
+              error_code_to_str(error_code::unknown_topic_or_partition)};
+            co_return co_await ctx.respond(std::move(resp));
+        }
+
+        ongoing_topic_reassignment topic_reassignment{.name = topic.name};
+
+        for (const int32_t pid : topic.partition_indexes) {
+            if (pid < 0) {
+                vlog(
+                  klog.error,
+                  "Invalid partition index: topic {}, pid {}",
+                  topic.name,
+                  pid);
+                resp.data.error_code = error_code::unknown_topic_or_partition;
+                resp.data.error_message = ss::sstring{
+                  error_code_to_str(error_code::unknown_topic_or_partition)};
+                co_return co_await ctx.respond(std::move(resp));
+            }
+
+            model::ntp ntp{
+              model::kafka_namespace, topic.name, model::partition_id{pid}};
+            auto reassignments
+              = co_await ctx.topics_frontend().list_partition_reassignments(
+                ntp, model::timeout_clock::now() + timeout_ms);
+            if (reassignments.error_code == error_code::none) {
+                vlog(
+                  klog.debug,
+                  "Reassignments: ntp {}, replicas {}, adding {}, removing {}",
+                  ntp,
+                  reassignments.replicas,
+                  reassignments.adding_replicas,
+                  reassignments.removing_replicas);
+                topic_reassignment.partitions.push_back(
+                  ongoing_partition_reassignment{
+                    .partition_index = model::partition_id{pid},
+                    .replicas = std::move(reassignments.replicas),
+                    .adding_replicas = std::move(reassignments.adding_replicas),
+                    .removing_replicas = std::move(
+                      reassignments.removing_replicas)});
+            } else {
+                vlog(
+                  klog.error,
+                  "Unknown problem listing reassignments: ntp {}, ec {} - {}",
+                  ntp,
+                  reassignments.error_code,
+                  reassignments.error_message);
+                resp_error = reassignments.error_code;
+                resp_error_message = reassignments.error_message;
+                break;
```

Comment:
Same issue here as in the other request handler, we can't exit early. All items in the request input list must be processed and one response per request in the input list is necessary to be returned to the client.